scripts/koji/koji_build.py

```diff
@@ -79,8 +79,15 @@ def push_bumped_release(git_repo, trigram):
             subprocess.check_call(['git', 'push', '--delete', 'origin', b])
         spec_path = glob('SPECS/*.spec')[0]
         with Specfile(spec_path) as spec:
-            # TODO: check koji build to use another final number when needed
-            spec.release = f'{spec.release}.0.{trigram}.1'
+            # find the next build number
+            package = spec_path.split('/')[-1][:-len('.spec')]
+            builds = subprocess.check_output(['koji', 'list-builds', '--package', package]).decode('utf8').splitlines()[2:]
+            base_name = f'{package}-{spec.version}-{spec.release}.0.{trigram}.'
+            base_name_re = re.escape(re.sub('%{.+}', "@@@", base_name)).replace('@@@', '.*') + r'(\d+)'
+            build_matchs = [re.match(base_name_re, b) for b in builds]
```

Comment:
that looks a bit complicated, what about just filtering on `startswith(basename)` and stripping the first `len(basename)` chars?

Reply:
base_name might include some macros, so it wouldn't match. So this is a (not so) complicated way to match without precisely knowing to what value the macro is expanded